### PR TITLE
feat: add new web lock api interface

### LINKED ISSUE #3456: Web Locks API?
I'm working on a project that could benefit from using the [Web Locks API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Locks_API) via web-sys.

I _think_ this API isn't currently available - is that right? If so, is it in-scope for the project / would this be a good issue for a first-time contributor (me)?

### PR DESCRIPTION
**Pull request by Solo-steven**
fixes #3456.
this is the first time I have contributed to this repo, If there is some definition that is wrong, please tell me! thanks ~

### CODE REVIEW

**Review by daxpedda — CHANGES_REQUESTED** [review]
Not sure exactly what happened here, did you check the CI failure?
Could you take a look at [the documentation](https://github.com/rustwasm/wasm-bindgen/blob/main/guide/src/contributing/web-sys/supporting-more-web-apis.md) and see if that generates the code correctly?

- Looking at the [reference](https://w3c.github.io/web-locks/#idl-index) there are a couple of APIs still missing.
- A changelog entry is missing.

It would also be nice if you link your sources. E.g. double-checking the [Mozilla implementation](https://hg.mozilla.org/mozilla-central/file/e91d172728590586e38bc431f3dc779479e365d5/dom/webidl/LockManager.webidl) to see if any functions are throwing.

Otherwise LGTM!

**Comment by Solo-steven** [comment]
Hi @daxpedda, thanks for your review, It is my fault that I wrote the `gen_*` file instead of using the auto codegen command, Now I have switched to auto codegen, and All the CI tests are passed.
Since MDN only have `LockManager` and `Lock` document, I only add the MDN link to `gen_lock` and `gen_lockManager`, if there is any wrong, please tell me ~ thanks ~

**Comment by daxpedda** [comment]
> Since MDN only have `LockManager` and `Lock` document, I only add the MDN link to `gen_lock` and `gen_lockManager`, if there is any wrong, please tell me ~ thanks ~

I only meant posting it here in the issue! But that's for the next time, I already got the links and posted them above :wink:.

**Review by daxpedda — CHANGES_REQUESTED** [review]
LGTM!
Changelog entry is missing.

**Comment by Solo-steven** [comment]
Hi @daxpedda thanks for your reply, but I can't find the document about adding Changelog, could you help me find the link to the document or give me some simple instructions? thanks ~ ( and sorry for bother you again )

**Comment by Solo-steven** [comment]
Hi @daxpedda  thanks for the reminder, I added the entry to the changelog, if the entry description is not correct, please tell me, thanks ~

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 ### Added
 
+* Add bindings for `WebLock` related api to `web-sys`.
```
Comment (daxpedda) [inline CHANGELOG.md:8]:
```suggestion
* Add bindings for the Web Locks API to `web-sys`.
```

**Review by daxpedda — APPROVED** [review]
Thanks!

**Pull request merged**